hashicorp/boundary-ui
# Icu 15181 add pagination to session recordings

# Description

This adds pagination support for Session Recordings along with search for `session-recording.id` and filters for user and target.

Another change that is made in this PR is how we handle redirects for the `scopes.scope.session-recordings.session-recording.channels-by-connection.channel` route. If the user has entered a URL where the session recording ID is wrong, we redirect the user back to the list view and display a toast message.


:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-15181)

## Screenshots (if appropriate)
Search and filter:

https://github.com/user-attachments/assets/8c748e27-898f-4443-a978-e01f571824c7


Redirect change:

https://github.com/user-attachments/assets/41eb6cde-7e64-4c94-96dc-619da173683a

## How to Test

Create sessions with targets that have recording enabled. After creating a few, you should see the table of session recordings with new filters above.

## Checklist
